tensorboard/components/tensor_widget/BUILD

```diff
@@ -1,18 +1,58 @@
 package(default_visibility = ["//tensorboard:internal"])
 
+load("@build_bazel_rules_nodejs//:defs.bzl", "rollup_bundle")
+load("@npm_bazel_jasmine//:index.bzl", "jasmine_node_test")
+load("@npm_bazel_typescript//:index.bzl", "ts_library")
+
 load("//tensorboard/defs:web.bzl", "tf_web_library")
 
 licenses(["notice"])  # Apache 2.0
 
 tf_web_library(
     name = "tensor_widget",
     srcs = [
+        ":tensor_widget_binary.js",
+    ],
+    path = "/tensor-widget",
+    deps = [
+        ":tensor_widget_binary",
+    ],
+)
+
+rollup_bundle(
+    name = "tensor_widget_binary",
+    entry_point = ":tensor-widget.ts",
+    deps = [
+        ":tensor_widget_lib",
+    ],
+)
+
+ts_library(
+    name = "tensor_widget_lib",
+    srcs = [
+        "dtype-utils.ts",
         "health-pill-types.ts",
-        "tensor-widget.html",
         "tensor-widget.ts",
         "types.ts",
     ],
-    path = "/tensor-widget",
-    visibility = ["//visibility:public"],
-    deps = [],
+)
+
+ts_library(
+    name = "test_lib",
+    testonly = True,
+    srcs = [
+        "dtype-utils-test.ts",
+    ],
+    deps = [
+        ":tensor_widget_lib",
+        "@npm//@types/chai",
+        "@npm//@types/jasmine",
```

Comment:
maybe not but you _may_ be able to get rid of this: https://github.com/tensorflow/tensorboard/blob/master/tsconfig-test.json#L4

Reply:
I verified that I can't get rid of it. If I removed it, errors like the following will occur:

> tensorboard/components/tensor_widget/dtype-utils-test.ts:20:1 - error TS2593: Cannot find name 'describe'. Do you need to install type definitions for a test runner? Try `npm i @types/jest` or `npm i @types/mocha` and then add `jest` or `mocha` to the types field in your tsconfig.
> 
> 20 describe('isIntegerDType', () => {
>

Reply:
Alright. Thanks for trying. I wonder why we need the tsconfig-test then :thinking: